src/rime/gear/selector.cc

```diff
@@ -252,6 +253,23 @@ bool Selector::End(Context* ctx) {
   return Home(ctx);
 }
 
+bool Selector::HiliteCandidateAt(Context* ctx, int index) {
+  Composition& comp = ctx->composition();
+  if (comp.empty())
+    return false;
+  int page_size = engine_->schema()->page_size();
+  if (index >= page_size)
+    return false;
+  int selected_index = comp.back().selected_index;
+  int page_start = (selected_index / page_size) * page_size;
+  // hilite an already hilited candidate -> select this candidate
+  if (page_start + index == selected_index)
+    return ctx->Select(selected_index);
+  comp.back().selected_index = page_start + index;
+  comp.back().tags.insert("paging");
```

Comment:
Highlighting a candidate does not necessarily change page.

Reply:
> Highlighting a candidate does not necessarily change page.

But this is consistent with the behavior of navigators—even if there's no turning pages, user is no longer actively composing which means paging keys should take priority over composing keys (e.g. in the case of `,` and `.`).